### PR TITLE
Add build error handling

### PR DESCRIPTION
**Pull request by trusktr**
**Summary**

Add error handling to build processes so they don't silently fail.

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix?
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

---

* [ ] DO NOT include files inside `lib` directory.

### CODE REVIEW

**Comment by trusktr** [comment]
Now the error I'm getting, which I don't understand, is

```
❯ node --trace-warnings ./node_modules/.bin/npm run build

> docsify@4.11.3 build /home/trusktr/src/lume+umbrella/packages/docsifyjs+docsify
> rimraf lib themes && run-s build:js build:css build:css:min build:ssr build:cover


> docsify@4.11.3 build:js /home/trusktr/src/lume+umbrella/packages/docsifyjs+docsify
> cross-env NODE_ENV=production node build/build.js

lib/plugins/ga.min.js
lib/plugins/ga.js
lib/plugins/matomo.min.js
lib/plugins/matomo.js
lib/plugins/emoji.min.js
lib/plugins/emoji.js
lib/plugins/external-script.js
lib/plugins/external-script.min.js
lib/plugins/disqus.js
lib/plugins/disqus.min.js
lib/plugins/gitalk.js
lib/plugins/gitalk.min.js
lib/plugins/zoom-image.min.js
lib/plugins/zoom-image.js
lib/plugins/search.js
lib/plugins/search.min.js
lib/plugins/front-matter.js
lib/plugins/front-matter.min.js
lib/docsify.js
Error: You must supply "output.name" for IIFE bundles.
    at error (/home/trusktr/src/lume+umbrella/packages/docsifyjs+docsify/node_modules/rollup/dist/rollup.js:10162:30)
    at iife (/home/trusktr/src/lume+umbrella/packages/docsifyjs+docsify/node_modules/rollup/dist/rollup.js:10408:9)
    at Chunk$1.render (/home/trusktr/src/lume+umbrella/packages/docsifyjs+docsify/node_modules/rollup/dist/rollup.js:15561:29)
    at /home/trusktr/src/lume+umbrella/packages/docsifyjs+docsify/node_modules/rollup/dist/rollup.js:18025:38
    at Array.map (<anonymous>)
    at /home/trusktr/src/lume+umbrella/packages/docsifyjs+docsify/node_modules/rollup/dist/rollup.js:18023:46
    at Generator.next (<anonymous>)
    at fulfilled (/home/trusktr/src/lume+umbrella/packages/docsifyjs+docsify/node_modules/rollup/dist/rollup.js:40:28)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
```

It fails on building `docsify.js`.

**Comment by trusktr** [comment]
But why would this fail now and not before?

File: build/build.js [hunk 1]
```diff
@@ -64,56 +67,68 @@ const buildAllPlugin = function () {
     {name: 'gitalk', input: 'gitalk.js'}
   ]
 
-  plugins.forEach(item => {
-    build({
+  const promises = plugins.map(item => {
+    return build({
```
Comment (anikethsaha) [inline build/build.js:71]: should it be `push` ?

**Comment by anikethsaha** [comment]
> But why would this fail now and not before?

not sure, I think rollup is async by default ?

File: build/build.js (hunk 1 above)
Comment (trusktr) [reply build/build.js:71]: This one's a `.map`, so it returns the whole array.

**Comment by trusktr** [comment]
The issue is basically

- https://github.com/rollup/rollup-plugin-commonjs/issues/274
- https://github.com/rollup/rollup/pull/3181

It's late though, so I will think about this tomorrow. I think that we'll just export whatever the globals we need from the index.js file, and set the `output.name`. This will replace the call to `initGlobalAPI`, plus make the code more useful. And avoiding globals is better.

**Comment by trusktr** [comment]
> rollup is async by default 

Yep, it is. In earlier versions of Node (f.e. Node 10) unhandled Promise rejections are silent errors that don't terminate the program. So we have to add the explicit error handling to ensure that the build fails when there's an error. In newer versions of Node, any unhandled Promise rejections automatically end the program.

File: build/build.js (hunk 1 above)
Comment (anikethsaha) [reply build/build.js:71]: I meant this can be with like `forEach + push` right !

this looks good too 👍

**Pull request merged**